src/Storefront/Resources/views/storefront/component/line-item/element/total-price.html.twig

```diff
@@ -9,14 +9,14 @@
         <div class="line-item-total-price-value">
             {# Shipping costs discounts always have a price of 0, which might be confusing, therefore we do not show those #}
             {% if lineItem.payload.discountScope != 'delivery' %}
-                {{ lineItem.price.totalPrice|currency }}{{ 'general.star'|trans|sw_sanitize }}
+                {{ lineItem.price.totalPrice|currency(order.currency.isoCode ?? context.currency.isoCode) }}{{ 'general.star'|trans|sw_sanitize }}
```

Comment:
Hey @Schrank,

I would prefer when we handle this as a "template-depdency" and the currency to use will be provided from the outer one.

```
{% sw_include '..../line-item/element/total-price.html.twig' with {
    lineItem: lineItem,
    currency: order.currency ?? context.currency
} %}
```

I bet this will be not the place where the price is included, i guess you have to pass through all templates, starting where the `order` variable is started first

Reply:
Good idea!

Reply:
In my opinion it would be better to pass the currency from the template where we explicitly want the display to change (Storefront/Resources/views/storefront/page/account/order-history/order-detail-list.html.twig) and then only pass the currency variable through. Here we have many call on possibly not set variables, which we like to reduce.

Storefront/Resources/views/storefront/page/account/order-history/order-detail-list.html.twig
```
{% block page_account_order_item_detail_line_item %}
    {% sw_include '@Storefront/storefront/component/line-item/line-item.html.twig' with {
            displayMode: 'order',
            showRemoveButton: false,
           currencyCode: lineItem.order.currency.isoCode 
     } %}
{% endblock %}
```

all other includes:
```
{% sw_include '@Storefront/storefront/component/....' with {
        currencyCode: currencyCode ?? null
 } %}
```

also a change to the name `currencyCode` would be nice to avoid conflicts and be more clear of the type of variable.

Reply:
Thanks for the input!

The consequence of this is, that the currency isn't available anymore, so if you want to change something and not use the code, but the ISO code or the sign you need to overwrite the template one level above.
Imho this is a disadvantage.

What is your pro argument? (Beside easier to read code and a little less complexity?)

Reply:
I am not sure wether we don't talk about different changes. Why would the currency not be available enymore?

Reply:
If I pass down only `currency.isoCode` instead of `currency` inside the `with`, then the `currencyEntity` is not available.

So I'm talking:
```
    currency: order.currency ?? context.currency
vs
    currencyCode: context.currency.isoCode ?? null
```

Did I misunderstand you?

Oh, I think I'm talking stupid - the currencyEntity is not available at all, is it? That means, what you say is not: Replace the currencyEntity by currencyCode, but rename the variable, so other people don't make the same mistake as I just did? 😂 

So the big question: Fix the name of the variable or make the currency a currencyEntity? Any opinions on that? @ssltg @OliverSkroblin

Reply:
Yes, that was the intention. Just rename the variable. But to pass the whole currency entity would also be ok for me, then it could stay by that name and we have more options later.

Reply:
I would say: change it to the currencyEntity. After all it is about the price template and it is likely that we can make use of the currencyEntity there.